src/vmspawn/vmspawn.c

```diff
@@ -0,0 +1,419 @@
+                        break;
+
+                case ARG_QEMU_KVM:
+                        if (isempty(optarg))
+                                arg_qemu_kvm = -1;
+                        else {
+                                r = parse_boolean(optarg);
+                                if (r < 0)
+                                    return log_error_errno(r, "Failed to parse boolean from --qemu-kvm=%s: %m", optarg);
+                                arg_qemu_kvm = r;
+                        }
+                        break;
+
+                case ARG_QEMU_GUI:
+                        arg_qemu_gui = true;
+                        break;
+
+                case ARG_SECBOOT:
+                        if (isempty(optarg))
+                                arg_secboot = -1;
+                        else {
+                                r = parse_boolean(optarg);
+                                if (r < 0)
+                                    return log_error_errno(r, "Failed to parse boolean from --secboot=%s: %m", optarg);
+                                arg_secboot = r;
+                        }
+                        break;
+
+                case ARG_SET_CREDENTIAL: {
+                        r = machine_credential_set(&arg_credentials, &arg_n_credentials, optarg);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to set credential from %s: %m", optarg);
+                        arg_settings_mask |= SETTING_CREDENTIALS;
+                        break;
+                }
+
+                case ARG_LOAD_CREDENTIAL: {
+                        r = machine_credential_load(&arg_credentials, &arg_n_credentials, optarg);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to load credential from %s: %m", optarg);
+
+                        arg_settings_mask |= SETTING_CREDENTIALS;
+                        break;
+                }
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        if (argc > optind) {
+                strv_free(arg_parameters);
+                arg_parameters = strv_copy(argv + optind);
+                if (!arg_parameters)
+                        return log_oom();
+
+                arg_settings_mask |= SETTING_START_MODE;
+        }
+
+        return 1;
+}
+
+static int run_virtual_machine(void) {
+        _cleanup_(ovmf_config_freep) OvmfConfig *ovmf_config = NULL;
+        _cleanup_strv_free_ char **cmdline = NULL;
+        int r;
+
+        bool use_kvm = arg_qemu_kvm == 1;
+        if (arg_qemu_kvm == -1) {
```

Comment:
when checking tristates please do checks via `< 0`, `> 0` and `== 0`. i.e. don't compare the literal values, but just the sign of things

Reply:
Fixed in c053f06da5.